shared/actions/provision.tsx

```diff
@@ -506,17 +506,31 @@ const showFinalErrorPage = (state: Container.TypedState, action: ProvisionGen.Sh
 
 const showUsernameEmailPage = () => RouteTreeGen.createNavigateAppend({path: ['username']})
 
-const forgotUsername = (_: Container.TypedState, action: ProvisionGen.ForgotUsernamePayload) =>
-  RPCTypes.accountRecoverUsernameWithEmailRpcPromise(
-    {email: action.payload.email},
-    Constants.forgotUsernameWaitingKey
-  )
-    .then(() => ProvisionGen.createForgotUsernameResult({result: 'success'}))
-    .catch(error =>
-      ProvisionGen.createForgotUsernameResult({
-        result: Constants.decodeForgotUsernameError(error),
-      })
+const forgotUsername = (_: Container.TypedState, action: ProvisionGen.ForgotUsernamePayload) => {
+  if (action.payload.email) {
+    return RPCTypes.accountRecoverUsernameWithEmailRpcPromise(
+      {email: action.payload.email},
+      Constants.forgotUsernameWaitingKey
+    )
+      .then(() => ProvisionGen.createForgotUsernameResult({result: 'success'}))
+      .catch(error =>
+        ProvisionGen.createForgotUsernameResult({
+          result: Constants.decodeForgotUsernameError(error),
+        })
+      )
+  } else {
```

Comment:
worth it to check here for a phone and error if there isn't one?

Reply:
It's protected by the component's logic, I'll explain this below.